### PR TITLE
WIP, ENH: Pypy update

### PR DESCRIPTION
**Pull request by mattip**
Try to minimize the special exceptions needed for PyPy

### CODE REVIEW

**Comment by mattip** [comment]
hmm, looks like there are lots of failures that need to be skipped/fixed in cython/fixed in PyPy.

File: Cython/Compiler/ModuleNode.py [hunk 1]
```diff
@@ -3097,11 +3097,7 @@ def generate_type_import_call(self, type, code, import_generator, error_code=Non
         if sizeof_objstruct != objstruct:
             if not condition:
                 code.putln("")  # start in new line
-            code.putln("#if defined(PYPY_VERSION_NUM) && PYPY_VERSION_NUM < 0x050B0000")
```
Comment (scoder) [inline Cython/Compiler/ModuleNode.py]: Doesn't this just remove backwards compatibility with PyPy 5 (however good or bad that was)?

File: Cython/Includes/cpython/datetime.pxd [hunk 2]
```diff
@@ -6,16 +6,16 @@ cdef extern from "Python.h":
 
 cdef extern from "datetime.h":
 
-    ctypedef extern class datetime.date[object PyDateTime_Date]:
+    ctypedef extern class datetime.date[object PyDateTime_Date, check_size ignore]:
```
Comment (scoder) [inline Cython/Includes/cpython/datetime.pxd]: I understand that this is nice for PyPy, but it removes a safety feature in CPython. A warning seems an acceptable price to pay, at least to me.

**Comment by mattip** [comment]
Closing. Will try again once github actions are activated.

**Pull request closed without merging**